pyads/pyads_ex.py

```diff
@@ -176,6 +177,33 @@ def adsAddRoute(net_id: SAmsNetId, ip_address: str) -> None:
         raise ADSError(error_code)
 
 
+def send_raw_udp_message(
+    ip_address: str, message: bytes, expected_return_length: int
+) -> Tuple[bytes, str]:
+    """Send a raw UDP message to the PLC and return the response
+
+    :param str ip_address: ip address of the PLC
+    :param message: the message to send to the PLC
+    :expected_return_length: number of bytes to expect in response
+    """
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_DGRAM)) as sock:  # UDP
+        # Listen on 55189 for the response from the PLC
+        sock.bind(("", 55189))
```

Comment:
Just out of curiosity: Is every TwinCAT plc sending data from 55189 and receiving on 48899?

Reply:
How about error handling? What happens if you can not bind to that port?

Reply:
For both of these, I just copied from #75. Regarding port 55189 -- the sending port doesn't actually matter, but the receiving port does, because that's where [ADS expects UDP data to be sent to](https://infosys.beckhoff.com/content/1033/tcremoteaccess/html/tcremoteaccess_firewall.html?id=4860001493050562785). I removed the port 55189 requirement and instead changed it to [bind to any available port using `0`](https://stackoverflow.com/a/1365284/4409369), and I updated the tests to reflect this as well.